apps/server/src/modules/deletion/controller/api-test/deletion-request-create.api.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { ExecutionContext, INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { ApiValidationError } from '@shared/common';
+import { Request } from 'express';
+import request from 'supertest';
+import { AuthGuard } from '@nestjs/passport';
+import { EntityManager } from '@mikro-orm/mongodb';
+import { AdminApiServerTestModule } from '../../../server/admin-api.server.module';
+import { DeletionRequestBodyProps, DeletionRequestResponse } from '../dto';
+import { DeletionDomainModel } from '../../domain/types';
+import { DeletionRequestEntity } from '../../entity';
+
+const baseRouteName = '/deletionRequests';
+
+class API {
+	app: INestApplication;
+
+	constructor(app: INestApplication) {
+		this.app = app;
+	}
+
+	async post(requestBody: object) {
+		const response = await request(this.app.getHttpServer())
+			.post(`${baseRouteName}`)
+			.set('Accept', 'application/json')
+			.send(requestBody);
+
+		return {
+			result: response.body as DeletionRequestResponse,
+			error: response.body as ApiValidationError,
+			status: response.status,
+		};
+	}
+}
+
+describe(`deletionRequest create (api)`, () => {
+	let app: INestApplication;
+	let em: EntityManager;
+	let api: API;
+	const API_KEY = '1ab2c3d4e5f61ab2c3d4e5f6';
+
+	beforeAll(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [AdminApiServerTestModule],
+		})
+			.overrideGuard(AuthGuard('api-key'))
+			.useValue({
+				canActivate(context: ExecutionContext) {
```

Comment:
This is the legacy hack to do api test. Please check examples with  { TestApiClient } from '@shared/testing';

Reply:
Should I change this if it is planned to change the authorisation for this API?